addons/l10n_es_edi_tbai/tests/common.py

```diff
@@ -0,0 +1,239 @@
+      <DescripcionFactura>manual</DescripcionFactura>
+      <DetallesFactura>
+        <IDDetalleFactura>
+          <DescripcionDetalle>producta</DescripcionDetalle>
+          <Cantidad>5.00</Cantidad>
+          <ImporteUnitario>1000.00</ImporteUnitario>
+          <Descuento>1000.00</Descuento>
+          <ImporteTotal>4840.00</ImporteTotal>
+        </IDDetalleFactura>
+      </DetallesFactura>
+      <ImporteTotalFactura>4840.00</ImporteTotalFactura>
+      <Claves>
+        <IDClave>
+          <ClaveRegimenIvaOpTrascendencia>01</ClaveRegimenIvaOpTrascendencia>
+        </IDClave>
+      </Claves>
+    </DatosFactura>
+    <TipoDesglose>
+      <DesgloseTipoOperacion>
+        <Entrega>
+          <Sujeta>
+            <NoExenta>
+              <DetalleNoExenta>
+                <TipoNoExenta>S1</TipoNoExenta>
+                <DesgloseIVA>
+                  <DetalleIVA>
+                    <BaseImponible>4000.00</BaseImponible>
+                    <TipoImpositivo>21.00</TipoImpositivo>
+                    <CuotaImpuesto>840.00</CuotaImpuesto>
+                    <OperacionEnRecargoDeEquivalenciaORegimenSimplificado>N</OperacionEnRecargoDeEquivalenciaORegimenSimplificado>
+                  </DetalleIVA>
+                </DesgloseIVA>
+              </DetalleNoExenta>
+            </NoExenta>
+          </Sujeta>
+        </Entrega>
+      </DesgloseTipoOperacion>
+    </TipoDesglose>
+  </Factura>
+  <HuellaTBAI>
+    <Software>
+      <LicenciaTBAI>___ignore___</LicenciaTBAI>
+      <EntidadDesarrolladora>
+        <NIF>___ignore___</NIF>
+      </EntidadDesarrolladora>
+      <Nombre>___ignore___</Nombre>
+      <Version>___ignore___</Version>
+    </Software>
+    <NumSerieDispositivo>___ignore___</NumSerieDispositivo>
+  </HuellaTBAI>
+</T:TicketBai>
+""".encode("utf-8")
+
+    L10N_ES_TBAI_SAMPLE_XML_CANCEL = """<T:AnulaTicketBai xmlns:T="urn:ticketbai:anulacion">
+  <Cabecera>
+    <IDVersionTBAI>1.2</IDVersionTBAI>
+  </Cabecera>
+  <IDFactura>
+    <Emisor>
+      <NIF>09760433S</NIF>
+      <ApellidosNombreRazonSocial>EUS Company</ApellidosNombreRazonSocial>
+    </Emisor>
+    <CabeceraFactura>
+      <SerieFactura>INVTEST</SerieFactura>
+      <NumFactura>01</NumFactura>
+      <FechaExpedicionFactura>01-01-2022</FechaExpedicionFactura>
+    </CabeceraFactura>
+  </IDFactura>
+  <HuellaTBAI>
+    <Software>
+      <LicenciaTBAI>___ignore___</LicenciaTBAI>
+      <EntidadDesarrolladora>
+        <NIF>___ignore___</NIF>
+      </EntidadDesarrolladora>
+      <Nombre>___ignore___</Nombre>
+      <Version>___ignore___</Version>
+    </Software>
+    <NumSerieDispositivo>___ignore___</NumSerieDispositivo>
+  </HuellaTBAI>
+</T:AnulaTicketBai>""".encode("utf-8")
+
+# ===== SII PATCHING =====
+# Adapt the SII common test class to include the refund reason on invoice creation
+# TODO find a cleaner way to do this ? (not clear yet if SII and TBAI sometimes have to send the same invoices)
+def create_invoice(cls, **kwargs):
```

Comment:
@odoostan Why not just create a specific method for tbai instead of patching the SII one? (e.g. create_invoice_tbai) and you put one in the common as well...  I tried to remove this and the tests seem to pass, so if ok for you, just remove.

Reply:
Very weird. We do have a method for TBAI. We patch SII's methods because the tests used to fail without it. I did re-test everything and can confirm this code is useless now.